clang/lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -94,10 +94,14 @@ void visualstudio::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   // If the VC environment hasn't been configured (perhaps because the user
   // did not run vcvarsall), try to build a consistent link environment.  If
   // the environment variable is set however, assume the user knows what
-  // they're doing. If the user passes /vctoolsdir or /winsdkdir, trust that
-  // over env vars.
-  if (const Arg *A = Args.getLastArg(options::OPT__SLASH_diasdkdir,
-                                     options::OPT__SLASH_winsysroot)) {
+  // they're doing. If the user passes /vctoolsdir or /winsdkdir or any of the
+  // other Windows SDK options, trust that over env vars.
+  const Arg *A = Args.getLastArg(options::OPT__SLASH_vctoolsdir,
```

Comment:
I think that we don't care about the value itself, so you should use `hasArg` over `getLastArg`.